.travis.yml

```diff
@@ -208,10 +212,23 @@ jobs:
             travis_terminate 1
           fi
         - npx github-release-from-cc-changelog $TRAVIS_TAG
+        - npm run pkg:upload
       deploy:
         provider: npm
         email: services@serverless.com
         on:
           tags: true
         api_key:
           secure: EgoetjrRlGfvGnmVp8A0btr1CzB0hl7owVIpbfk4zXJzhGEbHoVu0CG0IdmyLN+JlaZa7EDJTjkDCd6g3fVAh9TT7ZCeaq8YwbZDrql7mAJj7xYQAyM4eSkc95BRzcFJBx7Mxr6H90IDLxKr6ZtB+HEdiHN+59XbepKYYJeb1jHfnKn5xzOqk4BdnZo6pKfudfeO+7/BwJJ0FwlFA40bY2HS/Lp+NG/2IedNR7k3m/5W83/XH5qlWP8jhBKlxrAzks27aNo+42xHkRCVyPViJKq0mfz1hl2bfswChWHgaCuajp+0amNL39pgIX9eXxFc3bNX9Iftox5t31elEhsw06vvuAaVkKEd+VEMaDySbQ9M+irKZeREg+NFYZLnc2WiEE3Sexo6hm9eM2q2KEZ7bleN9B0CQAut1XXLRQEts80rzss4Z2Q7AZb9cOYBQlj9Wf1X0Y74UqvnDn83a4Y38a+lhx7J2q691ZeM1UFSCdO0QfeJRkB55bSyHqUqrLAqUN7eNsKGdBH0kvYIGFREgGgReEpBRAuNqWuJ/5qexp63Kbf+09raG5IvfxSIM5fJ5KE5VxSduBdRnSH0GNKfjuq296/Rg4fmm/bygZ3Yk5L6Wd41SUU8uHzlZFBwtcvxAKDTQe6s+5JU24ilqxOx6J4Ut34X3dIbLLAmoB1ogdM=
+
+    - stage: Publish to Chocolatey
```

Comment:
Is there any way we can make the publishing atomic? Could this be run if another step in the publishing process (see above) fails?

Same question applies if there's a possibility to publish to GiHub but fail to publish to Chocolatey.

Reply:
The way it's configured now is:

- Build binaries
- (only if all below succeeds) Publish version to npm
- (only if all below succeeds) Upload binaries for newly published version
- (only if all below succeeds) Publish to Chocolatey

If last fails, there's no attempt to revert any previous steps (also note it's impossible to unpublish version that was successfully published to npm)

I constructed it as that, intentionaly, as I believe we should not publish new binaries until we have confirmed npm publication (I would treat npm registry as core source of truth on which versions are published or not)

Then upload of successfully built binaries step is pretty simple, quite unlikely to fail, and even if it happens it should be straightforward to recover (we'll have CI failed status, which we should not ignore).

Same with chocolatey publish, it's quite straightforward, and if it fails, it needs to be investigated, and same publish command can be run by developer locally.

I also don't think it's harmful if for some moment standalone distribution will be delayed against npm

Reply:
That sounds good. I was not too familiar with the implementation and just wanted to raise awareness that there might be some dependencies here we should know about.

The reasoning totally makes sense. Thanks for doing the in-depth writeup :+1: